app/controllers/categories_controller.rb

```diff
@@ -1,58 +1,56 @@
-class CategoriesController < ApplicationController
-
-before_action :require_admin, except: [:index, :show]
-
-
-def index
-@categories = Category.paginate(page: params[:page], per_page: 5)
-end
-
-def new
-@category = Category.new
-end
-
-
-def create
-@category = Category.new(category_params)
-if @category.save
-  flash[:success] = "Category was created successfully"
-  redirect_to categories_path
-
-else
-  render 'new'
-end
-end
-
-
-def edit
-@category = Category.find(params[:id])
-end
-
-def update
-@category = Category.find(params[:id])
-if @category.update(category_params)
-flash[:success] = "Category name was successfully updated"
-redirect_to category_path(@category)
-else
-  render 'edit'
-end
-end
-
-def show
-@category = Category.find(params[:id])
-@category_articles = @category.articles.paginate(page: params[:page], per_page: 5)
-end
-
-private
-def category_params
-params.require(:category).permit(:name)
-end
-
-def require_admin
-if !logged_in? || (logged_in? and !current_user.admin?)
-  flash[:danger] = "Only admins can perform that action"
-  redirect_to categories_path
-end
-end
+# frozen_string_literal: true
 
+class CategoriesController < ApplicationController
```

Comment:
We need to authorize these controller methods as well except `index` and `show`. The methods should only be accessible to admins.

Reply:
ok